pkg/assembler/backends/ent/dependency.go

```diff
@@ -64,25 +68,38 @@ func (e DependencyEdges) PackageOrErr() (*PackageVersion, error) {
 	return nil, &NotLoadedError{edge: "package"}
 }
 
-// DependentPackageOrErr returns the DependentPackage value or an error if the edge
+// DependentPackageNameOrErr returns the DependentPackageName value or an error if the edge
 // was not loaded in eager-loading, or loaded but was not found.
-func (e DependencyEdges) DependentPackageOrErr() (*PackageName, error) {
+func (e DependencyEdges) DependentPackageNameOrErr() (*PackageName, error) {
 	if e.loadedTypes[1] {
-		if e.DependentPackage == nil {
+		if e.DependentPackageName == nil {
 			// Edge was loaded but was not found.
 			return nil, &NotFoundError{label: packagename.Label}
 		}
-		return e.DependentPackage, nil
+		return e.DependentPackageName, nil
 	}
-	return nil, &NotLoadedError{edge: "dependent_package"}
+	return nil, &NotLoadedError{edge: "dependent_package_name"}
+}
+
+// DependentPackageVersionOrErr returns the DependentPackageVersion value or an error if the edge
+// was not loaded in eager-loading, or loaded but was not found.
+func (e DependencyEdges) DependentPackageVersionOrErr() (*PackageVersion, error) {
+	if e.loadedTypes[2] {
```

Comment:
This seems very likely to cause issues on future refactoring as it is unclear why the element at index 2 is the one to be used here

Reply:
The same consideration as above applies here.